### PR TITLE
Use datetime.strptime tokens internally for parsing ISO 8601 week dates

### PR DESCRIPTION
**Pull request by yiransii**
## Pull Request Checklist

Thank you for taking the time to improve Arrow! Before submitting your pull request, please check all *appropriate* boxes:

- [ ] 🧪  Added **tests** for changed code.
- [x] 🛠️  All tests **pass** when run locally (run `tox` or `make test` to find out!).
- [x] 🧹  All linting checks **pass** when run locally (run `tox -e lint` or `make lint` to find out!).
- [ ] 📚  Updated **documentation** for changed code.
- [x] ⏩  Code is **up-to-date** with the `master` branch.

If you have *any* questions about your code changes or any of the points above, please submit your questions along with the pull request and we will try our best to help!

## Description of Changes

 Use datetime.strptime() tokens internally for parsing ISO 8601 week dates

## Questions:
1) ***tests*** My code has passed python3.7, 3.8, and 3.9 tests. How should I deal with the pypy3 issue and the python3.6 issue properly?
```
ERROR:   pypy3: could not install deps [-rrequirements.txt]; v = InvocationError('/Users/yiransi/Documents/opensource/arrow/.tox/pypy3/bin/python -m pip install -rrequirements.txt', 1)
SKIPPED:  py36: InterpreterNotFound: python3.6
  py37: commands succeeded
  py38: commands succeeded
  py39: commands succeeded
```
2) ***linting checks*** Should I remove those test cases if the linting checks are only failing with the following message?
```
- hook id: flake8
- exit code: 1

tests/test_arrow.py:341:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:349:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:362:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
tests/test_arrow.py:370:13: B015 Pointless comparison. This comparison does nothing but wastes CPU instructions. Remove it.
```

## Issue
Closes: #857

### CODE REVIEW

File: arrow/parser.py [hunk 1]
```diff
@@ -424,7 +424,11 @@ def _build_datetime(parts):
                 # day not given, default to 1
                 day = 1
 
-            dt = iso_to_gregorian(year, week, day)
+            date_string = f"{year}-{week}-{day}"
+
+            # %Y: year, %W: week number of the year(Monday as the first day of the week), %w: weekdate
+            dt = datetime.strptime(date_string, "%Y-%W-%w")
```
Comment (systemcatch) [inline arrow/parser.py:430]: I think we need to use the tokens for ISO 8601 weekdates here as they don't seem to be interchangeable (https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes), i.e. `"%G-%V-%u"`

Comment (systemcatch) [reply arrow/parser.py:430]: For example:

```shell
>>> from datetime import datetime
>>> dt1=datetime.strptime("2013-10-4", "%G-%V-%u")
>>> dt1
datetime.datetime(2013, 3, 7, 0, 0)
>>> dt2=datetime.strptime("2013-10-4", "%Y-%W-%w")
>>> dt2
datetime.datetime(2013, 3, 14, 0, 0)
```

**Review by systemcatch — CHANGES_REQUESTED** [review]
Hey @yiransii don't worry about the test stuff from flake8. For pypy3 and python 3.6 you can install them if you like but github actions is taking care of those tests.

File: arrow/parser.py (hunk 1 above)
Comment (yiransii) [reply arrow/parser.py]: I see. I have updated the branch so that it uses the ISO 8601 tokens. Thanks!

**Review by systemcatch — APPROVED** [review]
Nice we should be good to go here, thanks @yiransii

**Pull request merged**